nimare/generate.py

```diff
@@ -0,0 +1,269 @@
+"""Utilities for generating data for testing"""
+import os
+
+import numpy as np
+import nibabel as nib
+
+from .dataset import Dataset
+from .utils import get_resource_path
+from .meta.utils import compute_ma, get_ale_kernel
+from .transforms import vox2mm, mm2vox
+
+DEF_TINY = 1e-50
+
+
+def create_coordinate_dataset(
+    foci_num,
+    fwhm,
+    sample_size,
+    studies,
+    foci_coords=None,
+    foci_noise=0,
+    foci_weights=None,
+    rng=None,
+    sample_size_variance=None,
+):
+    """Generate coordinate based dataset for meta analysis.
+
+    Parameters
+    ----------
+    foci_num : :obj:`int` or :obj:`list`
+        The number of foci to be generated per study
+    fwhm : :obj:`float`
+        Full width at half maximum to define the probability
+        spread of the foci.
+    sample_size : :obj:`int` or :obj:`list`
+        Either mean number of participants in each study
+        or a list specifying the sample size for each
+        study.
+    studies : :obj:`int`
+        Number of studies to generate.
+    foci_coords : :obj:`list` of three item array_like objects (Optional)
+        Ground truth coordinates of foci.
+        If not specified, coordinates will be randomly selected.
+        If not the same length as the number of foci (i.e., `foci_num`),
+        foci will either be over-sampled or under-sampled.
+    foci_noise : :obj:`int` or :obj:`list`
+        Number of foci considered to be noise in each study
+        or a list of integers representing how many noise focis
+        there should be in each study.
+    foci_weights : :obj:`list` (Optional)
+        Weighing of each foci representing the probability of
+        that foci being sampled in a study.
+    rng : :class:`numpy.random.RandomState` (Optional)
+        Random state to reproducibly initialize random numbers.
+    sample_size_variance : :obj:`int` (Optional)
+        Variance of the number of participants in each study.
+        (must be >=0).
+
+    Returns
+    -------
+    ground_truth_foci : :obj:`list`
+        generated foci in xyz (mm) coordinates
+    dataset : :class:`nimare.Dataset`
+    """
+    if rng is None:
+        rng = np.random.RandomState(seed=1939)
```

Comment:
This could be controversial if people want to have more psuedo-random behavior, as this currently stands, if you call this function with the same inputs, you get the same output deterministically, so if you wanted to simulate a bunch of datasets with the same properties, you would need to pass in a different random state.

Another note, I could just have the user pass in a seed instead of an entire randomState object, that appears more reasonable.

Reply:
I don't see any problem with defaulting to a specific seed, although I think it would be better to have the seed be the default value in the actual function definition.